packages/core/src/BinaryData/FileSystem.manager.ts

```diff
@@ -78,6 +84,16 @@ export class FileSystemManager implements BinaryData.Manager {
 				await Promise.all([fs.rm(filePath), fs.rm(`${filePath}.metadata`)]);
 			}
 		}
+
+		// binary files stored in nested dirs - `filesystem-v2`
+
+		const workflowDirs = ids.map(({ workflowId }) => this.resolvePath(`workflows/${workflowId}`));
+
+		await Promise.all(
+			workflowDirs.map(async (dir) => {
+				await fs.rm(dir, { recursive: true });
+			}),
+		);
```

Comment:
Exactly. Right now we only store binary data here, but if in future we store also extra data here (execution runs, workflow history versions), those will be deleted as well.

Reply:
After discussion, let's revert [04adb51](https://github.com/n8n-io/n8n/pull/7307/commits/04adb517b19a64eb42208de56729e93f03e4a39d) so that `BinaryDataService` only touches binary data rather than dirs that can potentially contain other entities.